src/compile.rs

```diff
@@ -36,26 +50,208 @@ impl TypedProgram {
     /// Assumes that the input program has been correctly type-checked and **panics** if
     /// incompatible types are found that should have been caught by the type-checker.
     pub fn compile(&self, fn_name: &str) -> Result<(Circuit, &TypedFnDef), CompilerError> {
+        self.compile_with_constants(fn_name, HashMap::new())
```

Comment:
This made me wonder if there were other alternatives to optional arguments in Rust, so I ended up here:
https://hoverbear.org/blog/optional-arguments/
Your solution is the first one listed there as well :)